Autoconvert assets if browser supports it

Nitwel opened the pull request:
Fixes #16068
Fixes ENG-157
Ref: #15918

Nitwel committed:
init commit

Nitwel committed:
auto convert if browser supports it

Nitwel committed:
run linter

paescuj committed:
Merge branch 'main' into fix-16068

paescuj committed:
Add docs (moved from docs repo)

rijkvanzanten committed:
Merge branch 'main' into fix-16068

paescuj committed:
Merge branch 'main' into fix-16068

rijkvanzanten committed:
Merge branch 'main' into fix-16068

rijkvanzanten reviewed (changes requested):
Lets update this to be a `?format=auto` query param instead. I don’t think we need it to be an environment variable

Nitwel committed:
change to auto format query option

Nitwel commented:
Updated it to a query option.

paescuj committed:
Make linter happy :-)

paescuj committed:
Merge branch 'main' into fix-16068

paescuj committed:
Simply replace format when set to auto

The previous approach did not work because the format property was passed to sharp with the value "auto" and this is not a valid option

paescuj committed:
Quick 'n dirty appraoch to auto transform in app

paescuj commented:
Added a quick & dirty (but so far successful) approach in [7922e1c](https://github.com/directus/directus/pull/18012/commits/7922e1cd6f61924d5ad775e37d4fafab1c5f75a2) to auto transform images within the app as per https://github.com/directus/directus/issues/16068#issuecomment-1285849927.

@Nitwel Could you verify if the idea of allowing the `format=auto` option along with the `key` option makes sense and if so clean this pull request up so we can get it over the line?
Thanks! 😃

Nitwel committed:
update asset handling and support auto in presets

Nitwel commented:
It doesn't make sense to allow for `format=auto` when having a key set, as a key references a preset that contains the formatting. Meaning, we should change the presets instead of hacking our way around it. 😄 

But you brought up a good point which is that I did forgot a presets implementation for assets. Added that now.

rijkvanzanten committed:
Merge branch 'main' into fix-16068

rijkvanzanten commented:
@Nitwel @paescuj This is looking very promising so far! What's left TBD to get this review-ready / merged? 🙂

Nitwel commented:
Only a quick lint run 😅 and some testing but I'm happy where it's at right now.

paescuj committed:
Lint, lint, lint

paescuj committed:
Update api/src/controllers/assets.ts

paescuj commented on api/src/types/assets.ts:
```diff
@@ -65,21 +65,11 @@ export type TransformationMap = {
-
-export type TransformationPresetFormat = {
-	format?: 'jpg' | 'jpeg' | 'png' | 'webp' | 'tiff' | 'avif';
+	format?: 'auto' | 'jpg' | 'jpeg' | 'png' | 'webp' | 'tiff' | 'avif';
```
Small note on the side: While `auto` is now a valid input for presets on the user side, it must not be passed to `resolvePreset()`

paescuj committed:
Reuse types

paescuj committed:
DOn't mention choices in project settings page, already linked below

paescuj reviewed (approved):
Thanks @Nitwel  for giving it the finishing touch 😄
This is a great addition, especially from the app side 👍 
Nice clean-up, all my tests succeeded!
